packages/material-ui-styles/src/styled.js

```diff
@@ -8,14 +8,16 @@ import withStyles from './withStyles';
 function styled(Component) {
   const componentCreator = (style, options) => {
     function StyledComponent(props) {
-      const { classes, className, ...other } = props;
+      const { classes, className, component, ...other } = props;
+      const FinalComponent = component || Component;
```

Comment:
Why would you do this: 
```js
const StyledComponent = styled(({ component: Component, ...props}) => <Component {...props}>Hello, World!</Component>)({});
<StylecComponent component="div" />; // renders an empty div
```
At this point you might as well change the API to 
```js
const HelloWorld = ({ component: Component, ...props}) => <Component {...props}>Hello, World!</Component>;
const StyledComponent = ({});
<StylecComponent component={HelloWorld} />; // renders `Hello, World!`
```

Reply:
I'm revering